include/config_manager.h

```diff
@@ -0,0 +1,49 @@
+﻿#ifndef HARDWARE_CONFIG_H
+#define HARDWARE_CONFIG_H
+
+#include "configuration_enum.h"
+
+namespace rosflight_firmware
+{
+class ROSflight;
+class ConfigManager
+{
+public:
+  typedef struct __attribute__ ((packed))
+  {
+    uint32_t checksum;
+    hardware_config_t config[Configuration::DEVICE_COUNT];
+  } config_t;
+
+  static const int CONFIG_RESPONSE_MESSAGE_LENGTH{50};
+  typedef struct
+  {
+    bool successful;
+    bool reboot_required;
+    uint8_t message[CONFIG_RESPONSE_MESSAGE_LENGTH]; // Primarily for error messages
+  }config_response;
+
+  ConfigManager(ROSflight &RF, config_t &config);
+  // Reads the memory, and loads defaults if it is invalid. Call after the memory manager is ready
+  bool init();
+  bool configure_devices(); // Sends configurations to the board via the enable_device method
+  // Attempts to set a configuration, failing if the board config manager rejects it
+  config_response attempt_set_configuration(device_t device, uint8_t config);
+  // Sets a config without checks. This may cause an invalid configuration combo,
+  // so attempt_set_configuration is recommended
+  void set_configuration(device_t device, uint8_t config);
```

Comment:
It looks like this isn't used anywhere except in `attempt_set_configuration()`? Can we just fold it in to `attempt_set_configuration()` and get rid of this function? Or at the very least make this private?